demos/common/samples/args_helper.hpp

```diff
@@ -103,15 +103,18 @@ inline std::vector<std::string> split(const std::string &s, char delim) {
 }
 
 inline std::vector<std::string> parseDevices(const std::string& device_string) {
-    std::string comma_separated_devices = device_string;
-    const std::string::size_type colon_position = comma_separated_devices.find(":");
+    const std::string::size_type colon_position = device_string.find(":");
     if (colon_position != std::string::npos) {
-        comma_separated_devices = comma_separated_devices.substr(colon_position + 1);
+        std::string device_type = device_string.substr(0, colon_position);
+        if (device_type.compare("HETERO") == 0 || device_type.compare("MULTI") == 0) {
```

Comment:
```suggestion
        if (device_type == "HETERO" || device_type == "MULTI") {
```

Reply:
@rsreenu2000 Ping?

Reply:
Are there any more open questions to be addressed?

Reply:
I have suggested a change. You can either accept it or explain why you disagree.